tasks/collect_info_ent.yml

```diff
@@ -0,0 +1,55 @@
+---
+- block:
+  - name: Get registration token (RUN ONCE)
+    uri:
+      url: "{{ github_api_url }}/enterprises/{{ github_ent }}/actions/runners/registration-token"
+      headers:
+        Authorization: "token {{ access_token }}"
+        Accept: "application/vnd.github.v3+json"
+      method: POST
+      status_code: 201
+      force_basic_auth: yes
+    register: registration
+    run_once: yes
+    tags:
+      - install
+      - uninstall
+
+  - name: Check currently registered runners (RUN ONCE)
+    uri:
+      url: "{{ github_api_url }}/enterprises/{{ github_ent }}/actions/runners"
+      headers:
+        Authorization: "token {{ access_token }}"
+        Accept: "application/vnd.github.v3+json"
+      method: GET
+      status_code: 200
+      force_basic_auth: yes
+    check_mode: false
+    register: registered_runners
+    run_once: yes
+    tags:
+      - install
+      - uninstall
+
+  - name: Check service facts
+    service_facts:
+    tags:
+      - install
+      - uninstall
+
+  - name: Build service name
+    set_fact:
+      runner_service: "actions.runner.{{ ([( github_ent | default(github_account))[:45], runner_name] | join('.'))[:57]  }}.service"
```

Comment:
Could `github_account` just be set to the enterprise name in the `ent` context? 

If we'd never need to set both `github_ent` and `github_account` then we are adding to the potential for misconfiguration / misunderstanding for no reason. If there is a need to set both then that's fine. When is it required?

Reply:
Yes, no need for github_account when we are using the enterprise token.
I've removed this, and removed the assertion to supply the github_account at all when using an enterprise runner.